contracts/BaseStrategy.sol

```diff
@@ -258,42 +276,76 @@ abstract contract BaseStrategy {
      * would be negatively affected if `harvest()` is not called shortly, then this can return `true`
      * even if the keeper might be "at a loss" (keepers are always reimbursed by yEarn)
      *
+     * NOTE: `callCost` must be priced in terms of `want`
+     *
      * NOTE: this call and `tendTrigger` should never return `true` at the same time.
      */
-    function harvestTrigger(uint256 gasCost) public virtual view returns (bool);
+    function harvestTrigger(uint256 callCost) public view returns (bool) {
```

Comment:
Not sure if you saw my previous comment and I'm not great at github so adding it here. 
_"Debt is not always adjusted stepwise. For instance in an emergency shutdown situation. But even if it is strategy may not be able to free up 50k. But I suppose those edge cases can rewrite the whole method. Might be worth leaving virtual in the function declaration, that gives strategist the option to override if they have an edge case. Like you've done with tend trigger."_

For instance in the strategy I am writing at the moment I would want to overwrite to something like:
if (outstanding > 0 && crETH.getCash() > 0) return true;

To cover liquidity crisis situation on lending platform

Reply:
Emergency shutdown also shunts the debt to `totalDebt`, so it would also be a step adjustment.

But regardless, I think having it be `virtual` is a worthwhile improvement. Can't possibly account for _every_ condition in the template haha.

Reply:
I must be misunderstanding what you mean by step adjustment

Reply:
A step goes from `a` to `b ` instananeously (e.g. `y = b if t > 0 else a`)

A ramp goes from `a` to `b` incrementally (e.g. `y = m t + a`)

With the rate limit enabled, the total debt increases incrementally (ramps up) towards the debt limit. However, if the debt limit is lower than the total debt, it's called back all at once, letting the strategy decide the rate at which the debt is repaid.

Reply:
gotcha. I saw the debt rate but didn't realise it applied on the way out as well as the way in

Reply:
Not sure what is meant by "the way in", but the rate limit only applies when debt is being accumulated by strategies.

Where does that leave us w.r.t. this comment?